sparse_ho/algo/forward.py

```diff
@@ -127,9 +127,14 @@ def get_beta_jac_iterdiff(
             if verbose:
                 print("relative decrease = ", (pobj[-2] - pobj[-1]) / pobj0)
 
-        if use_stop_crit:
-            if (i > 1) and (pobj[-2] - pobj[-1] <= np.abs(pobj0 * tol)):
-                break
+        if use_stop_crit and i % Tgap == 0 and i > 0:
+            if hasattr(model, "_get_dobj"):
+                dobj = model._get_dobj(r, X, beta, alpha, y)
+                if pobj[-1] - dobj < np.abs(pobj0 * tol):
```

Comment:
Is the abs really needed ?

Reply:
Just in case the loss of one of our estimator is not positive

Reply:
It should not happen. Put an assert and remove the abs

Reply:
humm, I believe we put it here for the dual of the SVM, do you confirm @Klopfe 
This will be removed once we add a proper gap for the SVM too

Reply:
done